internal/guest/runtime/hcsv2/uvm.go

```diff
@@ -968,11 +969,29 @@ func modifyMappedVirtualDisk(
 		mountCtx, cancel := context.WithTimeout(ctx, time.Second*5)
 		defer cancel()
 		if mvd.MountPath != "" {
+			verityInfo := mvd.VerityInfo
 			if mvd.ReadOnly {
+				// The only time the policy is empty, and we want it to be empty
+				// is when no policy is provided, and we default to open door
+				// policy. In any other case, e.g. explicit open door or any
+				// other rego policy we would like to mount layers with verity.
+				if len(securityPolicy.EncodedSecurityPolicy()) > 0 {
+					mvd.GuestReadVerity = true
+				}
 				// containers only have read-only layers so only enforce for them
 				var deviceHash string
-				if mvd.VerityInfo != nil {
+				if verityInfo != nil {
 					deviceHash = mvd.VerityInfo.RootDigest
+				} else if mvd.GuestReadVerity {
+					devPath, err := scsi.GetDevicePath(ctx, mvd.Controller, mvd.Lun, mvd.Partition)
+					if err != nil {
+						return err
+					}
+					verityInfo, err = verity.ReadVeritySuperBlock(ctx, devPath)
```

Comment:
It's possible the device hasn't shown up yet at this point though, right?

Reply:
`GetDevicePath` waits until the device is  available I believe.

Reply:
Maybe it's worth moving the check for the dev node into `GetDevicePath`? What do you think?

Reply:
oh right. I misread that. yeah, I guess it'd make sense to move the check.